src/commit.c

```diff
@@ -382,41 +383,48 @@ int git_commit_amend(
 	return error;
 }
 
-int git_commit__parse(void *_commit, git_odb_object *odb_obj)
+int git_commit__parse_commit(git_commit *commit, git_odb_object *odb_obj, git_commit__parse_flags flags)
 {
-	git_commit *commit = _commit;
 	const char *buffer_start = git_odb_object_data(odb_obj), *buffer;
 	const char *buffer_end = buffer_start + git_odb_object_size(odb_obj);
-	git_oid parent_id;
 	size_t header_len;
 	git_signature dummy_sig;
+	git_oid parent_id;
+
+	assert(commit && odb_obj);
 
 	buffer = buffer_start;
 
+	/* The tree is always the first field */
+	if (flags & GIT_COMMIT_PARSE_ALL) {
+		if (git_oid__parse(&commit->tree_id, &buffer, buffer_end, "tree ") < 0)
+			goto bad_buffer;
+	} else {
+		buffer += strlen("tree ") + GIT_OID_HEXSZ + 1;
```

Comment:
Please check the length of `buffer` so that you make sure that you're not pointing off the end of it.  (Pointing out of bounds is UB, even if you don't dereference it.)